### ISSUE TITLE
Correct installation of files according to FHS

### ISSUE THREAD

**Issue by neurofreak-alt** [OpenBoard-org/OpenBoard#638]
Greetings!
I would like to see the structure of the project installed in the %buildroot brought closer to the FHS ( https://wiki.linuxfoundation.org/lsb/fhs) used in linux. Right now everything is built and installed in the ../product/release directory. I would like to see the configuration files in ../build/release/etc/OpenBoard, binary files in ..build/release/usr/bin, resources in ../build/release/usr/share/OpenBoard Will this happen in the future?

**Comment by letsfindaway** [comment]
Yes, I would also like to see that.

Currently in some places a set of patches is used to bring OpenBoard closer to the FHS layout. I'm also using patches for my [openSUSE builds of OpenBoard](https://build.opensuse.org/package/show/Education/OpenBoard). i think especially patch 0002-Install-to-correct-directories-on-linux.patch and 0004-Use-QStandardPaths-to-locate-resources.patch are important. I don't know exactly who started to create these patches, but might be they originate from the [Flathub builds](https://github.com/flathub/ch.openboard.OpenBoard/tree/master/patches).

The resulting layout is still not perfect, but at least

- the binary goes to `/usr/bin/openboard`,
- the library and all other resources files go to `/usr/share/OpenBoard`.

`/usr/share/OpenBoard/etc` is still a mess which would need some cleanup, but it is better than installing to `/opt`.

Note however that the deployment of OpenBoard at the schools in Geneva might have additional and contradicting requirements. I don't know, but it might be that it wouldn't be so easy for them to move to another directory layout. So this needs probably to be discussed and a migration strategy is needed.

**Comment by JBBgameich** [comment]
The patches used for flathub or just a `git format-patch upstream/1.7-dev` from [my repository on GitLab](https://gitlab.com/JBBgameich/OpenBoard/-/commits/1.7-dev), where I'm maintaining the patches for the flatpak packaging for each branch that I'm interested in in git.
If you want we can use that (or a new repository) and maintain these patches together.

**Comment by letsfindaway** [comment]
> The patches used for flathub or just a `git format-patch upstream/1.7-dev` from [my repository on GitLab](https://gitlab.com/JBBgameich/OpenBoard/-/commits/1.7-dev), where I'm maintaining the patches for the flatpak packaging for each branch that I'm interested in in git. If you want we can use that (or a new repository) and maintain these patches together.
> 
> EDIT: To be exact, the 1.7-dev branch is what the next flatpak release will be based on.

Good idea! My wish list for further evolution of these patches would be:

- Upstream as much as possible! Especially for Linux also the official packages are no longer bundling 3rd party libraries (see the new Ubuntu 22.04 package at https://github.com/OpenBoard-org/OpenBoard/releases/tag/v1.6.3). So there might be a chance to upstream the "Remove 3rd party" stuff?
- In this case also the move to `pkgconfig` could eventually be upstreamed.
- The onboard keyboard patch is also so obvious that it should get upstreamed! `checkSystemOnScreenKeyboardAvailable` could probably go to the `PlatformUtils`, but the mechanics of that seems to be useful.
- So the remainder are probably the correct installation paths and the related way to locate resources. That could be something where we could think how to make this even better and FHS compliant. And probably provide the necessary code to make it work on both the `/opt` based layout and the more FHS compliant layout.

But up to now I was merely using these patches and have not looked much deeper into them. Thanks for providing them!

**Comment by letsfindaway** [comment]
After merging #830, a `cmake` based build system creating a FHS compliant layout is now available in the `dev` branch. See https://github.com/letsfindaway/OpenBoard/wiki/CMake-builds for usage instructions.
